modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2205,46 +2205,12 @@ def map(
         PandasDataframe
             A new dataframe.
         """
-        if self.num_parts <= 1.5 * CpuCount.get():
```

Comment:
Why was the implementation moved to a lower level?

Reply:
It was moved here to avoid duplicating logic, and `map_partitions` in the partition manager is only used in these cases.

Reply:
Previously, your implementation was also used instead of `self._partition_mgr_cls.lazy_map_partitions` function, under some condition, but now it is not. Is that how it was intended?

Reply:
tree_reduce and groupby_reduce call map_partitions at the partition mgr. That's why @Retribution98 moved the logic there I guess.

Reply:
Yes, I see it. I think lazy map will be better in a lazy pipeline because some partitions may not be calculated further, so this issue is not relevant for this case.